core/market/decentralized/src/market.rs

```diff
@@ -32,6 +29,22 @@ pub enum MarketError {
     InternalError(#[from] ErrorMessage),
 }
 
+impl From<MarketError> for actix_web::HttpResponse {
+    fn from(e: MarketError) -> Self {
+        e.error_response()
+    }
+}
+
+impl ResponseError for MarketError {
+    fn error_response(&self) -> HttpResponse {
+        match self {
+            MarketError::Matcher(e) => e.error_response(),
+            MarketError::Negotiation(e) => e.error_response(),
+            MarketError::InternalError(e) => HttpResponse::InternalServerError().json(e),
+        }
+    }
+}
+
```

Comment:
Http stuff shouldn't leak into market stuff in my opinion.

Reply:
will move